### PR TITLE
added isEU

### PR DESCRIPTION
**Pull request by rm-Umar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,6 +104,16 @@ with non-English languages. `details.country` will still return country code.
 country = details.country # US
 country_name = details.country_name # United States
 ```
+##### EU Country
```
Comment (UmanShahzad) [inline README.md:107]: Newline separation:

```suggestion

##### European Union (EU) Country
```

File: README.md [hunk 2]
```diff
@@ -104,6 +104,16 @@ with non-English languages. `details.country` will still return country code.
...
+
+`details.is_eu` will return true if the country is a member of EU , as supplied by the `eu.json` file.
```
Comment (UmanShahzad) [inline README.md:109]:
```suggestion
`details.is_eu` will return `true` if the country is a member of the European Union (EU), as supplied by the `eu.json` file.
```

File: README.md [hunk 3]
```diff
@@ -104,6 +104,16 @@ with non-English languages. `details.country` will still return country code.
...
+
+```ruby
+is_eu = details.is_eu # false
+```
+It is possible to change the file by setting the `eucountries` setting when creating the `IPinfo` object.
```
Comment (UmanShahzad) [inline README.md:114]: Missing an extra newline for separation and let's please rename the setting itself:

```suggestion

It is possible to change the file by setting the `eu_countries` setting when creating the `IPinfo` object.
```

File: README.md [hunk 4]
```diff
@@ -104,6 +104,16 @@ with non-English languages. `details.country` will still return country code.
...
+
+The file must be a `.json` file with the following [structure.](lib/ipinfo/eu.json)
```
Comment (UmanShahzad) [inline README.md:116]:
```suggestion
The file must be a `.json` file with the [following structure](lib/ipinfo/eu.json).
```

File: lib/ipinfo.rb [hunk 5]
```diff
@@ -35,15 +37,19 @@ def initialize(access_token = nil, settings = {})
         maxsize = settings.fetch('maxsize', DEFAULT_CACHE_MAXSIZE)
         ttl = settings.fetch('ttl', DEFAULT_CACHE_TTL)
         @cache = settings.fetch('cache', DefaultCache.new(ttl, maxsize))
-        @countries = prepare_countries(settings.fetch('countries',
+        @countries = prepare_json(settings.fetch('countries',
                                                       DEFAULT_COUNTRY_FILE))
+        @eucountries = prepare_json(settings.fetch('eucountries',
```
Comment (UmanShahzad) [inline lib/ipinfo.rb:42]:
```suggestion
        @eu_countries = prepare_json(settings.fetch('eu_countries',
```

File: lib/ipinfo.rb [hunk 6]
```diff
@@ -14,6 +14,8 @@ module IPinfo
     DEFAULT_CACHE_TTL = 60 * 60 * 24
     DEFAULT_COUNTRY_FILE = File.join(File.dirname(__FILE__),
                                      'ipinfo/countries.json')
+    DEFAULT_EU_COUNTRY_FILE = File.join(File.dirname(__FILE__),
```
Comment (UmanShahzad) [inline lib/ipinfo.rb:17]:
```suggestion
    DEFAULT_EU_COUNTRIES_FILE = File.join(File.dirname(__FILE__),
```

File: lib/ipinfo.rb [hunk 7]
```diff
@@ -35,15 +37,19 @@ def initialize(access_token = nil, settings = {})
         maxsize = settings.fetch('maxsize', DEFAULT_CACHE_MAXSIZE)
         ttl = settings.fetch('ttl', DEFAULT_CACHE_TTL)
         @cache = settings.fetch('cache', DefaultCache.new(ttl, maxsize))
-        @countries = prepare_countries(settings.fetch('countries',
+        @countries = prepare_json(settings.fetch('countries',
                                                       DEFAULT_COUNTRY_FILE))
+        @eucountries = prepare_json(settings.fetch('eucountries',
+                                                      DEFAULT_EU_COUNTRY_FILE))
```
Comment (UmanShahzad) [inline lib/ipinfo.rb:43]:
```suggestion
                                                      DEFAULT_EU_COUNTRIES_FILE))
```

File: lib/ipinfo.rb [hunk 8]
```diff
@@ -35,15 +37,19 @@ def initialize(access_token = nil, settings = {})
...
     end
 
     def details(ip_address = nil)
         details = request_details(ip_address)
         if details.key? :country
             details[:country_name] =
                 @countries.fetch(details.fetch(:country), nil)
+            details[:is_eu] =
+                @eucountries.include?(details.fetch(:country))
```
Comment (UmanShahzad) [inline lib/ipinfo.rb:52]:
```suggestion
                @eu_countries.include?(details.fetch(:country))
```

**Review by UmanShahzad — APPROVED** [review]
Can you run the test cases one more time? I committed the suggestions. If all works let's merge.

**Comment by rm-Umar** [comment]
> Can you run the test cases one more time? I committed the suggestions. If all works let's merge.
Everything good
```bash 
$ ruby test/ipinfo_test.rb        
Started with run options --seed 12722

IPinfoTest
  test_lookup_ip6                                                 PASS (0.64s)
  test_lookup_ip4                                                 PASS (0.64s)
  test_set_adapter                                                PASS (0.00s)
  test_that_it_has_a_version_number                               PASS (0.00s)

Finished in 1.27761s
4 tests, 192 assertions, 0 failures, 0 errors, 0 skips
```

**Pull request merged**